frame/contracts/src/wasm/runtime.rs

```diff
@@ -393,6 +393,50 @@ define_env!(Env, <E: Ext>,
 		}
 	},
 
+	// Transfer some value to another account.
+	//
+	// If the value transfer was succesful zero is returned. Otherwise a non-zero value is returned.
+	// The scratch buffer is not touched. The receiver can be a plain account or a contract.
+	//
+	// - account_ptr: a pointer to the address of the beneficiary account
+	//   Should be decodable as an `T::AccountId`. Traps otherwise.
+	// - account_len: length of the address buffer.
+	// - value_ptr: a pointer to the buffer with value, how much value to send.
+	//   Should be decodable as a `T::Balance`. Traps otherwise.
+	// - value_len: length of the value buffer.
+	ext_transfer(
+		ctx,
+		account_ptr: u32,
+		account_len: u32,
+		value_ptr: u32,
+		value_len: u32
+	) -> u32 => {
+		let callee: <<E as Ext>::T as frame_system::Trait>::AccountId =
+			read_sandbox_memory_as(ctx, account_ptr, account_len)?;
+		let value: BalanceOf<<E as Ext>::T> =
+			read_sandbox_memory_as(ctx, value_ptr, value_len)?;
+
+		let ext = &mut ctx.ext;
+		let result = ctx.gas_meter.with_nested(ctx.gas_meter.gas_left(), |nested_meter| {
+			match nested_meter {
+				Some(nested_meter) => {
+					ext.transfer(
+						&callee,
+						value,
+						nested_meter,
+					)
+				},
+				// there is not enough gas to allocate for the nested call.
+				None => Err(sp_runtime::DispatchError::Other("Out of gas.")),
+			}
+		});
+
+		match result {
+			Ok(_) => Ok(STATUS_SUCCESS.into()),
+			Err(_) => Ok(TRAP_RETURN_CODE),
```

Comment:
Ah, weirdly, those constants are for the return type from the `call`/`deploy` function. If we wanted to avoid we would need to create a new set of constants I think to avoid mixing up different constants and change all the other usages (note e.g. `ext_get_storage` just above - it deals with unnamed constants). However, I am OK just leaving it with magic numbers as for now, we can get rid of the magic numbers at once in a following refactoring PRs.

Reply:
Having read the code closely, I think we don't even need errors here. 

A general strategy dealing with the out-of-gas (OOG) errors is to trap - generally because it is not feasible nor worthwhile to defend against out of gas with the exception of `call`s and `instantiate`s since they actually allow to devote gas budget.

Reply:
The return type is what I am particular uncertain about. There seems to be no consistent scheme around it.

Regarding your second comment: Do you think we should remove the return type and trap on any transfer error? There are more than OOG errors. These are the errors that can happen during a transfer:

> "not enough gas to pay transfer fee"
> "balance too low to send value"
> "value too low to create account"
> "destination balance too high to receive value"

You still want to trap?

Reply:
Result of a discussion with @pepyakin : We use explicit `0` and `1` as return values for now and re-iterate later when we refactor and unify error reporting. Already pushed that change.